### PR TITLE
tools: Fix layout handing in cephfs-data-scan (#13898)

### PR DESCRIPTION
**Pull request by jcsp**

### CODE REVIEW

**Comment by jcsp** [comment]
Tested by re-running TestDataScan and seeing that it passes.

**Comment by jcsp** [comment]
Here's a run with datascan passing and no failures attributable to this PR:
http://pulpito.ceph.com/jspray-2015-11-30_02:06:10-fs-wip-testing-jcsp---basic-multi/1163839

**Comment by jcsp** [comment]
Merge https://github.com/ceph/ceph-qa-suite/pull/737 at same time as this (that was on the testing branch used in the preceding run)

File: src/tools/cephfs/DataScan.cc [hunk 1]
```diff
@@ -616,6 +617,16 @@ int DataScan::scan_inodes()
       }
     } else {
       file_size = accum_res.ceiling_obj_size;
+      if (loaded_layout.fl_pg_pool == uint32_t(-1)
+          || loaded_layout.fl_object_size < accum_res.max_obj_size) {
+        // No layout loaded, or inconsistent layout, use default
+        guessed_layout = g_default_file_layout;
+        guessed_layout.fl_object_size = g_default_file_layout.fl_object_size;
+        guessed_layout.fl_stripe_unit = g_default_file_layout.fl_stripe_unit;
+        guessed_layout.fl_stripe_count = g_default_file_layout.fl_stripe_count;
```
Comment (ukernel) [inline src/tools/cephfs/DataScan.cc]: above three lines are useless

**Comment by jcsp** [comment]
Updated to remove redundant lines

**Pull request merged**